core/templates/dev/head/domain/exploration/ExplorationDraftObjectFactory.ts

```diff
@@ -17,54 +17,53 @@
  * domain objects.
  */
 
-var oppia = require('AppInit.ts').module;
-
-oppia.factory('ExplorationDraftObjectFactory', [
-  function() {
-    var ExplorationDraft = function(draftChanges, draftChangeListId) {
-      this.draftChanges = draftChanges;
-      this.draftChangeListId = draftChangeListId;
-    };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-    /**
-     * Checks whether the draft object has been overwritten by another
-     * draft which has been committed to the back-end. If the supplied draft id
-     * has a different value then a newer changeList must have been committed
-     * to the back-end.
-     * @param {Integer} - currentDraftId. The id of the draft changes whch was
-     *  retrieved from the back-end.
-     * @returns {Boolean} - True iff the currentDraftId is the same as the
-     * draftChangeListId corresponding to this draft.
-     */
-    ExplorationDraft.prototype.isValid = function(currentDraftId) {
-      return (currentDraftId === this.draftChangeListId);
-    };
-
-    ExplorationDraft.prototype.getChanges = function() {
-      return this.draftChanges;
-    };
+export class ExplorationDraft {
+  draftChanges: any;
+  draftChangeListId: any;
+  constructor(draftChanges, draftChangeListId) {
+    this.draftChanges = draftChanges;
+    this.draftChangeListId = draftChangeListId;
+  }
+  /**
+   * Checks whether the draft object has been overwritten by another
+   * draft which has been committed to the back-end. If the supplied draft id
+   * has a different value then a newer changeList must have been committed
+   * to the back-end.
+   * @param {Integer} - currentDraftId. The id of the draft changes whch was
+   *  retrieved from the back-end.
+   * @returns {Boolean} - True iff the currentDraftId is the same as the
+   * draftChangeListId corresponding to this draft.
+   */
+  isValid(currentDraftId) {
+    return (currentDraftId === this.draftChangeListId);
+  }
+  getChanges() {
+    return this.draftChanges;
+  }
+}
 
-    // TODO (ankita240796) Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable dot-notation */
-    ExplorationDraft['createFromLocalStorageDict'] = function(
-    /* eslint-enable dot-notation */
-        explorationDraftDict) {
-      return new ExplorationDraft(
-        explorationDraftDict.draftChanges,
-        explorationDraftDict.draftChangeListId);
+@Injectable({
```

Comment:
Ditto

Reply:
It is [here](https://github.com/oppia/oppia/pull/7145/files/04d70916960e2a621e192e22f8d3565501fef39e#diff-c73e82f4d4bf41cee7ec1807028d9c6bR47).